hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoCreateValidator.java

```diff
@@ -98,7 +98,7 @@ public boolean tooManyAutoAssociations(
             @NonNull final LedgerConfig ledgerConfig,
             @NonNull final EntitiesConfig entitiesConfig,
             @NonNull final TokensConfig tokensConfig) {
-        return n > ledgerConfig.maxAutoAssociations()
-                || (entitiesConfig.limitTokenAssociations() && n > tokensConfig.maxPerAccount());
+        return (entitiesConfig.limitTokenAssociations() && n > tokensConfig.maxPerAccount())
+                || (!entitiesConfig.unlimitedAutoAssociations() && n > ledgerConfig.maxAutoAssociations());
```

Comment:
Why do we need to check `!entitiesConfig.unlimitedAutoAssociations()` this here ? Even if the configuration is enabled, we should not be able to set a higher number. That configuration is just allowed for `-1` to be set in `maxAutoAssociations` right ?

Reply:
From my understanding we should not limit the automatic associations if the configuration is enabled.
Or is it only if the value is -1 we have unlimited auto associations and we keep limiting the positive values e.g. 5000 is valid but 5001 - invalid?
@Neeharika-Sompalli @MiroslavGatsanoga  Can you provide more info on this?

Reply:
I agree. If we allow unlimited auto associations, it is hard to explain why setting a concrete number should be limited. But this changes the current behavior and therefore we have to double-check (and eventually update the HIP).

Reply:
> Or is it only if the value is -1 we have unlimited auto associations and we keep limiting the positive values e.g. 5000 is valid but 5001 - invalid
Yes, that should be the intended behaviour. Having the feature flag enabled should allow `maxAutoAssociations` value for any account to be set to -1.
If an account has `maxAutoAssociations = -1` then we have unlimited auto associations for this account.